### ISSUE TITLE
after getting empty list from server the refetch method returns the cached results and not the empty list

### ISSUE THREAD

**Issue by stLoiz** [apollographql/apollo-client#7214]
**Intended outcome:**

I am using refetch() whenever the variables of the useQuery changes in order to filter the list. Every time a variable prop is changed it fetches the results according to the filters, when the results from the server are empty list [], the refetch() method returns the cached results and not the empty list [].

```
const { loading, error, data, fetchMore, refetch } = useQuery(Users, {
    variables: {
      userNumber: userNum,
    },
  });

useEffect(() => {
    if (userNum){
      refetch();
    } 
    }, [userNum]);
```
So its fetches the updated result every time `userNum ` prop changes with the list of users that have this `userNum `  however, if the list of users returned from the server is empty (as no user have this number),  I can see the previous cached results instead of an empty list. 

I have tried 
` fetchPolicy: 'cache-and-network',
    nextFetchPolicy: 'cache-first', `  
and still get the cached results instead of the empty list.

I have also tried the merge() in typePolicies and it worked! However when I apply the relayStylePagination it fetches again the cached results and not the empty list.

```const client = new ApolloClient({
  uri: 'http://localhost:5000/graphql',
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          devices: relayStylePagination(),
          merge(existing = {}, incoming) {
            return { ...existing, ...incoming };
          },
        },
      },
    },
  }),
});
```

My query is like this:
```query GetAllUsers(
    $after: ID
    $userNum: Int
    $dpiGt: String
    $dpiLt: String
  ) {
    users(
      after: $after
      userNumber: $userNum
      first: 30
      filter: { dpiGt: $dpiGt, dpiLt: $dpiLt }
    ) {
      edges {
        cursor
        node {
          id
          name
          userNumber
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
```

Is there  any bug that is causing this:  the cached results are not  updated if the list returned from the server is empty?

Same issue here  https://github.com/apollographql/apollo-client/issues/7195
**Versions**
System:
    OS: macOS 10.15.4
  Binaries:
    Node: 12.16.1 - ~/n/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.4 - ~/n/bin/npm
  Browsers:
    Chrome: 86.0.4240.80
    Firefox: 81.0.1
    Safari: 13.1
  npmPackages:
    @apollo/client: ^3.2.3 => 3.2.5

**Comment by hwillson** [comment]
Duplicate of https://github.com/apollographql/apollo-client/issues/7195.
